### PR TITLE
Added logic to update Teamcity real time with functional test results.

### PR DESCRIPTION
**Pull request by deepakaravindr**
1. This enables the test tab on Teamcity UI and updates it real time
   just like xunit test runner does with unit tests.
2. Refactored the scripts little bit and made it more robust.

@yishaigalatzer @alpaix @emgarten

### CODE REVIEW

**Comment by emgarten** [comment]
:shipit:

File: Scripts/NuGetFunctionalTestUtils.ps1 [hunk 1]
```diff
@@ -1,6 +1,59 @@
...
+{
+    param (
+    [Parameter(Mandatory=$true)]
+    [string]$singleResult)
+
+    $parts = $singleResult -split " "
+
+    if ($parts.Length -lt 3)
+    {
+        Write-Host -ForegroundColor Red 'WARNING: PARSING ISSUES. CANNOT WRITE TO TEAM CITY!'
```
Comment (alpaix) [inline Scripts/NuGetFunctionalTestUtils.ps1:27]: Write-Error with $singleResult to identify what's wrong

File: Scripts/NuGetFunctionalTestUtils.ps1 [hunk 2]
```diff
@@ -1,6 +1,59 @@
...
+        $testName = $parts[1];
+        $duration = $parts[2];
+
+        Write-Host "##teamcity[testStarted name='$testName']";
+
+        if ($status -eq "Failed")
+        {
+            if ($parts.Length -lt 4)
+            {
+               Write-Host -ForegroundColor Red 'WARNING: PARSING ISSUES. CANNOT WRITE TEST FAILURE TO TEAM CITY!'
```
Comment (alpaix) [inline Scripts/NuGetFunctionalTestUtils.ps1:42]: the same

**Comment by alpaix** [comment]
:ship:

File: Scripts/NuGetFunctionalTestUtils.ps1 (hunk 1 above)
Comment (deepakaravindr) [inline Scripts/NuGetFunctionalTestUtils.ps1:27]: Fixed in both places. Retaining it as Write-Host though

**Comment by deepakaravindr** [comment]
Merged with https://github.com/NuGet/NuGet.Client/commit/3dc939cd5fa8473608af18564b5e6166217af6f2

**Pull request closed without merging**